src/tests/unit/tests/bug-filing/azure-boards/azure-boards-bug-filing-service.test.tsx

```diff
@@ -0,0 +1,98 @@
+import { AzureBoardsSettingsForm } from '../../../../../bug-filing/azure-boards/azure-boards-settings-form';
+import { SettingsFormProps } from '../../../../../bug-filing/types/settings-form-props';
+import { UserConfigMessageCreator } from '../../../../../common/message-creators/user-config-message-creator';
+import { BugServicePropertiesMap } from '../../../../../common/types/store-data/user-configuration-store';
+
+describe('AzureBoardsBugFilingServiceTest', () => {
+    let userConfigMessageCreatorMock: IMock<UserConfigMessageCreator>;
+    let props: SettingsFormProps<AzureBoardsBugFilingSettings>;
+    let projectStub: string;
+    let issueDetailsLocationStub: AzureBoardsIssueDetailField;
+
+    const invalidTestSettings: AzureBoardsBugFilingSettings[] = [
+        null,
+        {} as AzureBoardsBugFilingSettings,
+        undefined,
+        { projectURL: '' } as AzureBoardsBugFilingSettings,
+        { projectURL: '', issueDetailsField: '' as AzureBoardsIssueDetailField },
+        { projectURL: 'some project', issueDetailsField: '' as AzureBoardsIssueDetailField },
+        { projectURL: '', issueDetailsField: 'some issue details location' as AzureBoardsIssueDetailField },
+    ];
+
+    beforeEach(() => {
+        projectStub = 'some project';
+        issueDetailsLocationStub = 'some location' as AzureBoardsIssueDetailField;
+        userConfigMessageCreatorMock = Mock.ofType(UserConfigMessageCreator);
+        props = {
+            deps: {
+                userConfigMessageCreator: userConfigMessageCreatorMock.object,
+            },
+            settings: {
+                projectURL: 'some project',
+                issueDetailsField: 'some location' as AzureBoardsIssueDetailField,
+            },
+        };
+    });
+
+    it('static properties', () => {
+        expect(AzureBoardsBugFilingService.key).toBe('azureBoards');
+        expect(AzureBoardsBugFilingService.displayName).toBe('Azure Boards');
+        expect(AzureBoardsBugFilingService.isHidden).toBeUndefined();
+    });
+
+    it('buildStoreData', () => {
+        const expectedStoreData: AzureBoardsBugFilingSettings = {
+            projectURL: projectStub,
+            issueDetailsField: issueDetailsLocationStub,
+        };
+        expect(AzureBoardsBugFilingService.buildStoreData(projectStub, issueDetailsLocationStub)).toEqual(expectedStoreData);
+    });
+
+    it('getSettingsFromStoreData', () => {
+        const expectedStoreData: AzureBoardsBugFilingSettings = {
+            projectURL: projectStub,
+            issueDetailsField: issueDetailsLocationStub,
+        };
+        const givenData: BugServicePropertiesMap = {
+            'some other service': {},
+            [AzureBoardsBugFilingService.key]: expectedStoreData,
+        };
+        expect(AzureBoardsBugFilingService.getSettingsFromStoreData(givenData)).toEqual(expectedStoreData);
+    });
+
+    describe('check for invalid settings', () => {
```

Comment:
Dan's comment yesterday makes me think - would it be a little tidier to have a `describe('isSettingsValid')` around the `it.each` for invalid settings and an `it` for valid settings?

Reply:
Done

Reply:
Consider I didn't touch this file much. I only updated on test. Github shows the file is new because I change the containing folder name. So all this comments are basically not part of the original PR.